src/test/java/io/jenkins/plugins/httpclient/RobustHTTPClientTest.java

```diff
@@ -24,19 +24,110 @@
 
 package io.jenkins.plugins.httpclient;
 
-import static org.junit.Assert.*;
+import static org.hamcrest.MatcherAssert.*;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThrows;
 
+import hudson.AbortException;
+import hudson.model.TaskListener;
+import java.io.File;
+import java.io.IOException;
 import java.net.URL;
+import java.nio.file.Files;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.ClassRule;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.WithoutJenkins;
 
 public class RobustHTTPClientTest {
 
+    @ClassRule
+    public static JenkinsRule j = new JenkinsRule();
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
     @Test
+    @WithoutJenkins
     public void sanitize() throws Exception {
-        assertEquals("http://x.com/some/path", RobustHTTPClient.sanitize(new URL("http://x.com/some/path")));
-        assertEquals(
-                "https://x.com/some/path?…", RobustHTTPClient.sanitize(new URL("https://x.com/some/path?auth=s3cr3t")));
-        assertEquals(
-                "https://…@x.com/otherpath", RobustHTTPClient.sanitize(new URL("https://user:s3cr3t@x.com/otherpath")));
+        assertThat(
+                RobustHTTPClient.sanitize(new URL("http://example.com/some/long/path")),
+                is("http://example.com/some/long/path"));
+        assertThat(
+                RobustHTTPClient.sanitize(new URL("https://example.com/some/path?auth=s3cr3t")),
+                is("https://example.com/some/path?…"));
+        assertThat(
+                RobustHTTPClient.sanitize(new URL("https://user:s3cr3t@example.com/otherpath")),
+                is("https://…@example.com/otherpath"));
+    }
+
+    @Test
+    @WithoutJenkins
+    public void sanitizeThrowsException() {
+        final AssertionError e = assertThrows(AssertionError.class, () -> {
+            RobustHTTPClient.sanitize(new URL("https://example.com/ /has/space/in/url/"));
```

Comment:
This is misleading. https://github.com/jenkinsci/apache-httpcomponents-client-4-api-plugin/blob/master/src/main/java/io/jenkins/plugins/httpclient/RobustHTTPClient.java#L261-L264 only throws an assertion error when assertions are enabled, which they would not be in production. Perhaps better to just switch this to a warning. (Unfortunately `URL` allows bogus strings which `URI` does not.)

Reply:
Thanks.  I've added two tests in b2a588a99f04c84bd83e0c22eebc0ade60ccdc54, one with assertions enabled and one with assertions disabled.  I've confirmed that the tests pass as expected whether assertions are enabled or disabled.